micrometer-observation/src/main/java/io/micrometer/observation/Observation.java

```diff
@@ -309,13 +310,13 @@ static Observation createNotStarted(ObservationConvention<Context> observationCo
      */
     static <T extends Context> Observation createNotStarted(ObservationConvention<T> observationConvention,
             Supplier<T> contextSupplier, ObservationRegistry registry) {
-        if (registry == null || registry.isNoop() || observationConvention == NoopObservationConvention.INSTANCE) {
```

Comment:
`NoopObservationConvention` was never used and it is package-private, since no-one asked for this I removed it to simplify things.